frontend/src/components/ConstraintWarning.vue

```diff
@@ -0,0 +1,61 @@
+<!--
+SPDX-FileCopyrightText: 2020 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+<template>
+  <div v-if="value" :class="{ 'd-flex' : icon }">
+    <v-tooltip v-if="icon" max-width="800px" top>
+      <template v-slot:activator="{ on: tooltip }">
+        <v-icon
+        v-on="tooltip"
+        :small="small"
+        class="pr-1"
+        color="warning"
+        >mdi-alert-circle-outline</v-icon>
+      </template>
+      <div>{{constraintTitle}}</div>
+      <div>{{constraintMessage}}</div>
+    </v-tooltip>
+    <v-alert v-else type="warning" outlined>
+      <div class="font-weight-bold">{{constraintTitle}}</div>
+      <div>{{constraintMessage}}</div>
+    </v-alert>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'ConstraintWarning',
+  props: {
+    value: {
+      type: Boolean
+    },
+    constraintType: {
```

Comment:
The name of the component ist `ConstraintWarning`. Why do we repeat the word `constraint` but `warning` in the property names. I would use `type` and `message`. Or better use a default slot for the message.